### PR TITLE
Add `DevServicesOracleDatasourceTestCase`

### LINKED ISSUE #16172: Dev-services and related tests for Oracle
## Description

Follow-up on #15964 : that PR introduces support for Oracle's JDBC driver, but lacks the corresponding dev-services, as well as the corresponding tests (which, crucially, also test that Agroal's `ExceptionSorter` is correctly set).

## Implementation ideas

See `io.quarkus.jdbc.db2.deployment.DevServicesDB2DatasourceTestCase`: basically, we need the same test in the jdbc-oracle extension.

The implementation should be rather similar to what we did for DB2.

### PR DESCRIPTION
**Pull request by famod**
Follow up to https://github.com/quarkusio/quarkus/pull/23465#issuecomment-1030921515

The test itself is pretty much c&p from other dev services tests.

Fix #16172

### CODE REVIEW

**Comment by famod** [comment]
From `JVM Tests - JDK 11`:
```
2022-02-07T00:09:49.7869926Z [INFO] -------------------------------------------------------
2022-02-07T00:09:49.7870997Z [INFO]  T E S T S
2022-02-07T00:09:49.7871698Z [INFO] -------------------------------------------------------
2022-02-07T00:09:51.6688870Z [INFO] Running io.quarkus.jdbc.oracle.deployment.DevServicesOracleDatasourceTestCase
2022-02-07T00:11:57.8967393Z 2022-02-07 00:09:57,304 WARN  [org.tes.uti.TestcontainersConfiguration] (build-61) Attempted to read Testcontainers configuration file at file:/home/runner/.testcontainers.properties but the file was not found. Exception message: FileNotFoundException: /home/runner/.testcontainers.properties (No such file or directory)
2022-02-07T00:11:57.9009171Z 2022-02-07 00:09:58,510 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-61) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
2022-02-07T00:11:57.9011782Z 2022-02-07 00:09:58,521 INFO  [org.tes.DockerClientFactory] (build-61) Docker host IP address is localhost
2022-02-07T00:11:57.9020462Z 2022-02-07 00:09:58,586 INFO  [org.tes.DockerClientFactory] (build-61) Connected to docker: 
2022-02-07T00:11:57.9024099Z   Server Version: 20.10.11+azure-3
2022-02-07T00:11:57.9024490Z   API Version: 1.41
2022-02-07T00:11:57.9028652Z   Operating System: Ubuntu 20.04.3 LTS
2022-02-07T00:11:57.9028979Z   Total Memory: 6947 MB
2022-02-07T00:11:57.9034090Z 2022-02-07 00:09:58,593 INFO  [org.tes.uti.ImageNameSubstitutor] (build-61) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2022-02-07T00:11:57.9035198Z 2022-02-07 00:10:00,081 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Starting to pull image
2022-02-07T00:11:57.9036094Z 2022-02-07 00:10:00,117 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
2022-02-07T00:11:57.9041903Z 2022-02-07 00:10:00,811 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  2 pending,  1 downloaded,  0 extracted, (56 KB/? MB)
2022-02-07T00:11:57.9042916Z 2022-02-07 00:10:00,813 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  1 pending,  2 downloaded,  0 extracted, (327 KB/? MB)
2022-02-07T00:11:57.9048708Z 2022-02-07 00:10:00,832 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  0 pending,  3 downloaded,  0 extracted, (327 KB/5 MB)
2022-02-07T00:11:57.9061482Z 2022-02-07 00:10:01,016 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  0 pending,  3 downloaded,  1 extracted, (2 MB/5 MB)
2022-02-07T00:11:57.9066834Z 2022-02-07 00:10:01,275 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  0 pending,  3 downloaded,  2 extracted, (2 MB/5 MB)
2022-02-07T00:11:57.9071543Z 2022-02-07 00:10:01,440 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pulling image layers:  0 pending,  3 downloaded,  3 extracted, (5 MB/5 MB)
2022-02-07T00:11:57.9076878Z 2022-02-07 00:10:01,452 INFO  [org.tes.DockerClientFactory] (docker-java-stream--975636617) Pull complete. 3 layers, pulled in 1s (downloaded 5 MB at 5 MB/s)
2022-02-07T00:11:57.9077845Z 2022-02-07 00:10:02,431 INFO  [org.tes.DockerClientFactory] (build-61) Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2022-02-07T00:11:57.9078577Z 2022-02-07 00:10:02,433 INFO  [org.tes.DockerClientFactory] (build-61) Checking the system...
2022-02-07T00:11:57.9444719Z 2022-02-07 00:10:02,434 INFO  [org.tes.DockerClientFactory] (build-61) ✔︎ Docker server version should be at least 1.6.0
2022-02-07T00:11:57.9473690Z 2022-02-07 00:10:02,572 INFO  [org.tes.DockerClientFactory] (build-61) ✔︎ Docker environment should have more than 2GB free disk space
2022-02-07T00:11:57.9495209Z 2022-02-07 00:10:02,638 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (build-61) Pulling docker image: docker.io/gvenzl/oracle-xe:21-slim. Please be patient; this may take some time but only needs to be done once.
2022-02-07T00:11:57.9504490Z 2022-02-07 00:10:03,965 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Starting to pull image
2022-02-07T00:11:57.9510742Z 2022-02-07 00:10:03,966 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
2022-02-07T00:11:57.9520055Z 2022-02-07 00:10:04,773 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Pulling image layers:  1 pending,  1 downloaded,  0 extracted, (75 MB/? MB)
2022-02-07T00:11:57.9526878Z 2022-02-07 00:10:07,939 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Pulling image layers:  1 pending,  1 downloaded,  1 extracted, (335 MB/? MB)
2022-02-07T00:11:57.9532145Z 2022-02-07 00:10:13,206 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Pulling image layers:  0 pending,  2 downloaded,  1 extracted, (895 MB/905 MB)
2022-02-07T00:11:57.9533353Z 2022-02-07 00:10:40,050 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Pulling image layers:  0 pending,  2 downloaded,  2 extracted, (905 MB/905 MB)
2022-02-07T00:11:57.9534332Z 2022-02-07 00:10:40,064 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (docker-java-stream-1755772379) Pull complete. 2 layers, pulled in 36s (downloaded 905 MB at 25 MB/s)
2022-02-07T00:11:57.9539362Z 2022-02-07 00:10:40,104 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (build-61) Creating container for image: docker.io/gvenzl/oracle-xe:21-slim
2022-02-07T00:11:57.9540362Z 2022-02-07 00:10:40,148 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (build-61) Container docker.io/gvenzl/oracle-xe:21-slim is starting: 0d8df7362a121643bd3cc1f2f1c258628c5fe14aaab39d73f3a7ff74d62d2b4c
2022-02-07T00:11:57.9545404Z 2022-02-07 00:11:55,818 INFO  [🐳 .io/gvenzl/oracle-xe:21-slim]] (build-61) Container docker.io/gvenzl/oracle-xe:21-slim started in PT1M53.203727S
2022-02-07T00:11:57.9546527Z 2022-02-07 00:11:55,818 INFO  [io.qua.dev.ora.dep.OracleDevServicesProcessor] (build-61) Dev Services for Oracle started.
2022-02-07T00:11:57.9547621Z 2022-02-07 00:11:55,827 INFO  [io.qua.dat.dep.dev.DevServicesDatasourceProcessor] (build-61) Dev Services for the default datasource (oracle) started.
2022-02-07T00:11:57.9561688Z 2022-02-07 00:11:57,621 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT on JVM started in 1.527s. 
2022-02-07T00:11:57.9562391Z 2022-02-07 00:11:57,632 INFO  [io.quarkus] (main) Profile test activated. 
2022-02-07T00:11:57.9570055Z 2022-02-07 00:11:57,637 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, jdbc-oracle, narayana-jta, smallrye-context-propagation]
2022-02-07T00:12:01.9819761Z 2022-02-07 00:11:59,865 INFO  [io.quarkus] (main) Quarkus stopped in 0.054s
2022-02-07T00:12:02.4699418Z [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 130.337 s - in io.quarkus.jdbc.oracle.deployment.DevServicesOracleDatasourceTestCase
```

File: extensions/jdbc/jdbc-oracle/deployment/pom.xml [hunk 1]
```diff
@@ -49,6 +64,40 @@
...
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <!-- force the default, overriding the DevServices exlude -->
```
Comment (gsmet) [inline extensions/jdbc/jdbc-oracle/deployment/pom.xml:94]:
```suggestion
                                <!-- force the default, overriding the DevServices exclude -->
```

**Review by gsmet — APPROVED** [review]
Looks good except for the small typo.

**Review by Sanne — APPROVED** [review]
Thanks! I'll apply the same "skip rules" to other deployment modules.

**Pull request merged**